### PR TITLE
Remove references to --pull=true and --pull=false 

### PR DESCRIPTION
**Pull request by rhatdan**
#### Does this PR introduce a user-facing change?

```release-note
Fix documentation on default --pull value for podman build.
```

### CODE REVIEW

**Comment by edsantiago** [comment]
This seems like a pretty significant breaking change.

**Comment by rhatdan** [comment]
Only think different from my understanding is that always will attempt to pull even if the image exists in local storage, which is very wrong.  where as newer will pull only if the image in the registry is different then the local one.

Hard to see how this could break anyone. @nalind @mtrmac Does --pull always actually replace the content in containers storage with the same content again, or does it pull and then realize it already has the content and drop it.

**Comment by rhatdan** [comment]
Oops my mistake, went from pull=missing to pull=newer to match Docker defaults to missing, reverting that change

**Comment by mtrmac** [comment]
> An image is considered to be newer when the digests are different.

Per that, the `newer` is a just a lie.

---

I don’t immediately know what the options do in detail… it’s up to the author of the PR to understand the situation and advocate for a specific change.

c/image would _typically_ not re-pull layers already present locally. For the most part, “pulling exactly the image again” should be a cheap enough mostly-metadata operation, such that “optimizing that out” by making another check should, _ideally_, not make things faster. In practice, there are complications.

So, I think the difference between `newer` and `always` is basically these complications.

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL
This is now pretty much just a documentation change. PTAL.  
@giuseppe @nalind @mheon @mtrmac @flouthoc @baude @Luap99 PTAL

File: docs/source/markdown/options/pull.image.md [hunk 1]
```diff
@@ -4,9 +4,9 @@
 ####> are applicable to all of those.
 #### **--pull**=*policy*
 
-Pull image policy. The default is **always**.
+Pull image policy. The default is **missing**.
```
Comment (vrothberg) [inline docs/source/markdown/options/pull.image.md:7]: That doesn't look right, so let's bikeshed a bit more.

"missing" means Podman will not pull the image when it's in the storage.  "always" means Podman will attempt to pull the image - ignoring the storage - but may skip layers if they're already present.

"always" is the current behavior.

Comment (vrothberg) [reply docs/source/markdown/options/pull.image.md:7]: Note that the defaults may differ between the various commands.

Comment (vrothberg) [reply docs/source/markdown/options/pull.image.md:7]: So `podman pull` defaults to "always".  `podman run` defaults to "missing" etc.

Comment (vrothberg) [reply docs/source/markdown/options/pull.image.md:7]: So I think we need to remove mentioning the default in this shared file and mention it at the include sites of the specific commands.

Comment (Luap99) [reply docs/source/markdown/options/pull.image.md:7]: see the top line in this file, these docs here are only for podman build and farm build NOT run and create

Comment (vrothberg) [reply docs/source/markdown/options/pull.image.md:7]: Thanks! LGTM

**Review by vrothberg — APPROVED** [review]
LGTM

**Review by mtrmac — COMMENTED** [review]
The implementation of the change LGTM.

As for the goals of the change, can we have a written record of the rationale for no longer documenting `true`/`false`?

AFAICS the options continue to work (good); and from a 1-minute grep `docker build` seems to have a boolean `--pull` option.

So, is this to align this option with some general Podman policy WRT Docker-compatibility options (there may well be one, I don‘t know), or is this a special case worth a comment in the code?

**Review by flouthoc — APPROVED** [review]
LGTM

**Comment by rhatdan** [comment]
Just for simplification of the code.

**Pull request merged**